rbac/management/workspace/view.py

```diff
@@ -103,77 +89,20 @@ def list(self, request, *args, **kwargs):
+        """
+        return super().destroy(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        """Delegate to service for destroy logic."""
+        self._service.destroy(instance)
 
+    @transaction.atomic()
     def update(self, request, *args, **kwargs):
         """Update a workspace."""
-        self.validate_workspace(request, "put")
-        self.update_validation(request)
-        return super().update(request=request, args=args, kwargs=kwargs)
-
-    def partial_update(self, request, *args, **kwargs):
-        """Patch a workspace."""
-        self.validate_workspace(request, "patch")
-        payload = request.data or {}
-        for field in payload:
-            if field not in VALID_PATCH_FIELDS:
-                message = f"Field '{field}' is not supported. Please use one or more of: {VALID_PATCH_FIELDS}."
-                error = {"workspace": [_(message)]}
-                raise serializers.ValidationError(error)
-
-        self.update_validation(request)
-
-        return super().update(request=request, args=args, kwargs=kwargs)
-
-    def update_validation(self, request):
-        """Validate a workspace for update."""
-        instance = self.get_object()
-        parent_id = request.data.get("parent_id")
-        if str(instance.id) == parent_id:
-            message = "Parent ID and ID can't be same"
-            error = {"workspace": [_(message)]}
-            raise serializers.ValidationError(error)
-
-    def validate_required_fields(self, request, required_fields):
-        """Validate required fields for workspace."""
-        for field in required_fields:
-            if field not in request.data:
-                message = f"Field '{field}' is required."
-                error = {"workspace": [_(message)]}
-                raise serializers.ValidationError(error)
-
-    def validate_workspace(self, request, action="create"):
-        """Validate a workspace."""
-        parent_id = request.data.get("parent_id")
-        tenant = request.tenant
-        workspace_type = request.data.get("type", Workspace.Types.STANDARD)
-        if workspace_type != Workspace.Types.STANDARD:
-            message = f"Only workspace type {Workspace.Types.STANDARD} is allowed."
-            error = {"workspace": [_(message)]}
-            raise serializers.ValidationError(error)
-        if action == "create":
-            self.validate_required_fields(request, REQUIRED_CREATE_FIELDS)
-        elif action == "put":
-            self.validate_required_fields(request, REQUIRED_PUT_FIELDS)
-        if parent_id:
-            validate_uuid(parent_id)
-            if not Workspace.objects.filter(id=parent_id, tenant=tenant).exists():
-                message = f"Parent workspace '{parent_id}' doesn't exist in tenant"
-                error = {"workspace": [message]}
-                raise serializers.ValidationError(error)
+        return super().update(request, *args, **kwargs)
```

Comment:
This allows for unpacking `kwargs` correctly, meaning we can rely on `partial_update=True` being passed to the serializer, meaning we can get rid of the patch serializer.